eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/UtxosSpec.hs

```diff
@@ -187,3 +180,70 @@ spec =
               CollectErrors
                 [BadTranslation . inject . InlineDatumsNotSupported @era $ TxOutFromInput lockedTxIn]
           ]
+
+    describe "Plutus gov policies for ParameterChange and TreasuryWithdrawal" $ do
+      it "ParameterChange" $ do
+        let pparamsUpdate = def & ppuCommitteeMinSizeL .~ SJust 1
+        let govAction = ParameterChange SNothing pparamsUpdate SNothing
+        invalidPlutusVersionGovPolicyTxFails govAction guessTheNumber3
+        invalidPlutusVersionGovPolicyTxFails govAction guessTheNumber3V2
+        alwaysSucceedsGovPolicyTxAcceptedValid govAction
+        alwaysFailsGovPolicyTxAcceptedInvalid govAction
+
+      it "TreasuryWithdrawal" $ do
+        rewardAccount <- registerRewardAccount
+        let withdrawals = Map.fromList [(rewardAccount, Coin 1000)]
+        let govAction = TreasuryWithdrawals withdrawals SNothing
+        invalidPlutusVersionGovPolicyTxFails govAction guessTheNumber3
+        invalidPlutusVersionGovPolicyTxFails govAction guessTheNumber3V2
+        alwaysSucceedsGovPolicyTxAcceptedValid govAction
+        alwaysFailsGovPolicyTxAcceptedInvalid govAction
+
+alwaysSucceedsGovPolicyTxAcceptedValid ::
+  forall era.
+  (HasCallStack, ConwayEraImp era) =>
+  GovAction era ->
+  ImpTestM era ()
+alwaysSucceedsGovPolicyTxAcceptedValid govAction = do
+  let plutus = alwaysSucceedsPlutus @'PlutusV3 3
+  setConstitutionScript $ hashPlutusScript plutus
+  proposal <- proposalWithPlutusGovPolicy govAction plutus
+  submitProposal_ proposal
```

Comment:
Only spend scripts have 3 arguments. Also the workflow should be adjusted:
* we don't want to overwrite the script hash, we can just accept a closure, eg `ParameterChange SNothing pparamsUpdate . SJust`. This will avoid using `error` in `error "Gov Policy tests are only...`
* We should be getting the ScriptHash by adding the script to the ImpTestState, this ensures we can use it
* we can use the regular `submitGovAction` function, no need to add extra helpers
```suggestion
alwaysSucceedsGovPolicyTxAcceptedValid ::
  (HasCallStack, ConwayEraImp era) =>
  (SciptHash (EraCrypto ers) -> GovAction era) ->
  ImpTestM era ()
alwaysSucceedsGovPolicyTxAcceptedValid mkGovAction = do
  scriptHash <-
    impAddPlutusScript $
      ScriptTestContext
        (alwaysSucceedsPlutus @'PlutusV3 2)
        PlutusArgs
          { paSpendDatum = Nothing
          , paData = P1.I 3
          }
  setConstitutionScript scriptHash
  submitGovAction_ $ mkGovAction scriptHash
```